src/react/account/AccountUpdateUser.tsx

```diff
@@ -0,0 +1,191 @@
+import { useDispatch, useSelector } from 'react-redux';
+import type { AppState } from '../../types/state';
+import { Banner } from '@scality/core-ui';
+import { Button } from '@scality/core-ui/dist/next';
+import Joi from '@hapi/joi';
+import { joiResolver } from '@hookform/resolvers';
+import { useForm } from 'react-hook-form';
+import { useOutsideClick } from '../utils/hooks';
+import { useIAMClient } from '../IAMProvider';
+import { queryClient } from '../App';
+import { useMutation } from 'react-query';
+import { useHistory, useParams } from 'react-router-dom';
+import { notFalsyTypeGuard } from '../../types/typeGuards';
+const regexpName = /^[\w+=,.@ -]+$/;
+const schema = Joi.object({
+  name: Joi.string()
+    .label('Name')
+    .required()
+    .min(2)
+    .max(64)
+    .regex(regexpName)
+    .message('Invalid Name'),
+});
+
+const AccountUpdateUser = () => {
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const IAMClient = useIAMClient();
+  const { IAMUserName, accountName } = useParams<{ IAMUserName: string, accountName: string }>();
+  const { register, handleSubmit, errors } = useForm({
+    resolver: joiResolver(schema)
+  });
+  const updateUserMutation = useMutation(
+    (newUserName: string) => {
+      const oldUserName = IAMUserName;
+      return notFalsyTypeGuard(IAMClient).updateUser(newUserName, oldUserName)
+        .then(() => {
+          queryClient.setQueryData(['listIAMUsers', accountName], (old: any) => {
+            if (old) {
+              const pages: any = old.pages;
+              pages.some((page: any) => {
+                const users = page.Users;
+                const index = users.findIndex(
+                  (user: any) => user.UserName === oldUserName,
+                );
+                return index !== -1 && (users[index].UserName = newUserName);
+              });
+              return {
+                ...old,
+                pages,
+              };
+            }
+          });
+        })
+        .catch((err) => {
+          dispatch(handleErrorMessage(`${err}`, 'byModal'));
+        })
+        .finally(() => {
+          dispatch(networkEnd());
+        });
+    },
+    {
+      onSuccess: () => {
+        history.goBack();
+      },
+      onError: () => {
+        const str = 'An error occurred during the user delete.';
+        dispatch(handleErrorMessage(str, 'byModal'));
+      },
+    },
+  );
+
+  /**
+   * This part has to be handle
+   */
+  const hasError = useSelector(
+    (state: AppState) =>
+      !!state.uiErrors.errorMsg && state.uiErrors.errorType === 'byComponent',
+  );
+  const errorMessage = useSelector(
+    (state: AppState) => state.uiErrors.errorMsg,
+  );
+  const loading = useSelector(
+    (state: AppState) => state.networkActivity.counter > 0,
+  );
+
+  const onSubmit = ({ name }: {name: string}) => {
+    clearServerError();
+    updateUserMutation.mutate(name);
+  };
+
+  const handleCancel = (e: any) => {
```

Comment:
```suggestion
  const handleCancel = (e: MouseEvent<HTMLButtonElement>) => {
```

So you will also need to add `MouseEvent` in React import.